src/rp2/plugin/accounting_method/hifo.py

```diff
@@ -80,13 +81,14 @@ def initialize(self, taxable_event_iterator: Iterator[AbstractTransaction], acqu
 
     # AVL tree node keys have this format: <spot_price>_<internal_id>. The internal_id part is needed to disambiguate transactions
     # that have the same spot_price. Internal_id is padded right in a string of fixed length (KEY_DISAMBIGUATOR_LENGTH).
+    # The highest internal_id is for the earliest acquired lot
     def _get_avl_node_key(self, spot_price: RP2Decimal, internal_id: str) -> str:
-        return f"{spot_price}_{internal_id:0>{self.KEY_DISAMBIGUATOR_LENGTH}}"
+        return f"{spot_price}_{(int(self.MAX_KEY_DISAMBIGUATOR)-int(internal_id)):0>{self.KEY_DISAMBIGUATOR_LENGTH}}"
```

Comment:
This is very clever, but it breaks abstraction layers: it assumes that `internal_id` is numeric and that it grows monotonically with time, both of which are not guaranteed. However your intuition is good: I think you can achieve the same effect (without introducing assumptions) by passing `str(index)` instead of `acquired_lot.internal_id` at line 70.

Reply:
Since it's no longer `internal_id`, can you rename the parameter to something like `node_id`?